projects/storefrontlib/cms-components/content/banner/banner.component.ts

```diff
@@ -25,4 +25,14 @@ export class BannerComponent {
   getTarget(data: CmsBannerComponent): string | null {
     return data.external === 'true' || data.external === true ? '_blank' : null;
   }
+
+  getImage(data: CmsBannerComponent): Image | ImageGroup | undefined {
```

Comment:
instead of doing this to make `type assert` into something else, maybe let's just include this use case to the typing of the container in `media.component.ts`.

I believe we should include `CmsBannerComponentMedia | CmsResponsiveBannerComponentMedia` in it as well. 
My reason for this is we are normalizing the media container and resolving it down the line in the data flow (from MediaService), so I think we shouldn't manipulate the type assertion like this since actual data that was passed is either `CmsBannerComponentMedia` or `CmsResponsiveBannerComponentMedia`. Only reason this is working is because those interfaces are optional and it contains the url property. 

what do you think?

Reply:
forget what I said. Let's merge it but a reminder to fix all typings during the strict mode work phase